### PR TITLE
feat: Add PODIO plugin, Track EDM backend 

### PR DESCRIPTION
**Pull request by paulgessinger**
This PR adds a self-contained PODIO plugin, independent from the EDM4hep plugin. It implements backends for the `TrackContainer` and `MultiTrajectory` EDM. Tests are included for conversion of both EDM types.

Blocked by:
- #2093 
- #2092 
- #2091 
- #2090

### CODE REVIEW

File: Plugins/Podio/src/PodioUtil.cpp [hunk 1]
```diff
@@ -78,7 +78,7 @@ ActsPodioEdm::Surface convertSurfaceToPodio(const ConversionHelper& helper,
     }
     result.boundValuesSize = values.size();
 
-    Eigen::Map<ActsSymMatrix<4>> trf{result.transform.data()};
```
Comment (asalzburger) [inline Plugins/Podio/src/PodioUtil.cpp]: Why is this part of this PR, was that not caught bey the CI?

**Review by asalzburger — APPROVED** [review]
Is the PodIO test not run in the ?

File: Plugins/Podio/src/PodioUtil.cpp (hunk 1 above)
Comment (paulgessinger) [reply Plugins/Podio/src/PodioUtil.cpp]: I just now updated this PR from a few weeks ago and had to update a few files.

**Comment by paulgessinger** [comment]
This adds a few unit tests, and I believe those should run, yes.

**Comment by paulgessinger** [comment]
This is green once again. @asalzburger can you have a look?

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6379192)!

|status|job|
|------|---|
|:red_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380871) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380870) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380869) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380868) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380867) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380866) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380865) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380864) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380863) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380862) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380861) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380860) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380859) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380858) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380857) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33380856) |

**Pull request merged**